src/main/scala/diplomaticobjectmodel/model/OMRTLModule.scala

```diff
@@ -29,22 +70,34 @@ trait Synchronous extends Synchronicity
 trait Asynchronous extends Synchronicity
 
 case class OMRTLReset(
+  name: String,
+  description: Option[String],
   activeEdge: Option[OMSignalAssertionLevel],
   clock: String, // This will always be the name of the clock signal on the to p-level module
   synchronicity: Option[Synchronicity]
-)
+) extends OMSignal
 
 case class OMResetVector(
+  name: String,
+  description: Option[String],
   width: Int
-)
+) extends OMSignal
+
+// Interrupts
+case class InterruptSignal(
+  name: String,
+  description: Option[String],
+  width: Int
+) extends OMSignal
 
 case class OMRTLInterface(
-  clocks: List[OMClock],
-  clockRelationships: List[OMClockRelationship],
-  resets: List[OMRTLReset]
+  clocks: Seq[OMClock],
+  clockRelationships: Seq[OMClockRelationship],
+  resets: () => Seq[OMRTLReset],
```

Comment:
Why is this a thunk that returns a `Seq[OMRTLReset]` rather than just a plain `Seq[OMRTLReset]`? All of the object model classes should just be plain data structures that don't involve function pointers.